spring-ai-core/src/main/java/org/springframework/ai/tool/resolution/SpringBeanToolCallbackResolver.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright 2023-2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.tool.resolution;
+
+import com.fasterxml.jackson.annotation.JsonClassDescription;
+import kotlin.jvm.functions.Function0;
+import kotlin.jvm.functions.Function1;
+import kotlin.jvm.functions.Function2;
+import org.springframework.ai.chat.model.ToolContext;
+import org.springframework.ai.tool.ToolCallback;
+import org.springframework.ai.tool.function.FunctionToolCallback;
+import org.springframework.ai.tool.util.ToolUtils;
+import org.springframework.ai.util.json.JsonSchemaGenerator;
+import org.springframework.ai.util.json.SchemaType;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Description;
+import org.springframework.context.support.GenericApplicationContext;
+import org.springframework.core.KotlinDetector;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.core.ResolvableType;
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * A Spring {@link ApplicationContext}-based implementation that provides a way to
+ * retrieve a bean from the Spring context and wrap it into a {@link ToolCallback}.
+ *
+ * @author Christian Tzolov
+ * @author Christopher Smith
+ * @author Sebastien Deleuze
+ * @author Thomas Vitale
+ * @since 1.0.0
+ */
+public class SpringBeanToolCallbackResolver implements ToolCallbackResolver {
```

Comment:
It's pretty much the same logic as DefaultFunctionCallbackResolver. I also added autotests since we didn't have any for this logic.